### PR TITLE
Move pkg/kubectl/version to staging

### PR DESCRIPTION
**Pull request by seans3**
* Moves `pkg/kubectl/version` to staging.
* Since this package affects the version variables that are compiled into `kubectl`, we update the make/BUILD files to point to the new package.
* Manually validated the version with the following commands:

`make clean; make WHAT=cmd/kubectl`
`bazel clean; bazel build //cmd/kubectl`

followed by checking the output of the following command using the resultant binary:

`kubectl version --client`

The output for both correctly looked like:
```
Client Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.0-alpha.1.340+79d153d8d157d6-dirty", GitCommit:"79d153d8d157d67c1cee5c1002f4e5d79934d299", GitTreeState:"dirty", BuildDate:"2019-07-26T22:45:49Z", GoVersion:"go1.12.7", Compiler:"gc", Platform:"linux/amd64"}
```

* Publishes this code to the new kubectl repository: https://github.com/kubernetes/kubectl/pkg/version.
* Code will be available for re-use after publishing.

/kind cleanup
/sig cli
/area kubectl
/area code-organization
/priority important-soon

```release-note
NONE
```

### CODE REVIEW

**Comment by dims** [comment]
/approve

(in principle!)

File: hack/lib/version.sh [hunk 1]
```diff
@@ -159,7 +159,7 @@ kube::version::ldflags() {
     ldflags+=(
       "-X '${KUBE_GO_PACKAGE}/pkg/version.${key}=${val}'"
       "-X '${KUBE_GO_PACKAGE}/vendor/k8s.io/client-go/pkg/version.${key}=${val}'"
-      "-X '${KUBE_GO_PACKAGE}/pkg/kubectl/version.${key}=${val}'"
+      "-X '${KUBE_GO_PACKAGE}/staging/src/k8s.io/kubectl/pkg/version.${key}=${val}'"
```
Comment (liggitt) [inline hack/lib/version.sh]: why does this use staging and client-go uses vendor? did you verify `kubectl version` returns the correct output?

**Comment by liggitt** [comment]
can you add a test to test-cmd that does something like `kubectl version --client -o json | jq .clientVersion.gitVersion` and verifies it does not contain the hard-coded `v0.0.0`?

File: hack/lib/version.sh (hunk 1 above)
Comment (seans3) [reply hack/lib/version.sh]: You were correct; the vendor version is what works. I also updated the `def.bzl` to use a `vendor` path. I validated using the following:

`make clean; make WHAT=cmd/kubectl`

then checking the resulting binary with:

`_output/local/bin/linux/amd64/kubectl version --client`

which showed the version had correctly been recorded.

```
Client Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.0-alpha.1.340+79d153d8d157d6-dirty", GitCommit:"79d153d8d157d67c1cee5c1002f4e5d79934d299", GitTreeState:"dirty", BuildDate:"2019-07-26T22:39:49Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
```

I also checked the bazel output with:

`bazel clean; bazel build //cmd/kubectl`

checking the resulting output at:

`bazel-bin/cmd/kubectl/linux_amd64_pure_stripped/kubectl version --client`

which showed:

```
Client Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.0-alpha.1.340+79d153d8d157d6-dirty", GitCommit:"79d153d8d157d67c1cee5c1002f4e5d79934d299", GitTreeState:"dirty", BuildDate:"2019-07-26T22:45:49Z", GoVersion:"go1.12.7", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by seans3** [comment]
> can you add a test to test-cmd that does something like `kubectl version --client -o json | jq .clientVersion.gitVersion` and verifies it does not contain the hard-coded `v0.0.0`?

Sure. I will look deeper, but I think we already have quite a few `kubectl version` integration tests in `run_kubectl_version_tests()` which is in `test/cmd/version.sh`.

**Comment by dims** [comment]
/lgtm
/hold

@seans3 please remove hold when you feel this is ready

**Comment by liggitt** [comment]
Yeah, there was an integration test that required client and server version to match that failed before

/hold cancel

**Pull request merged**